user_tools/docs/tools_e2e_tests.md

```diff
@@ -0,0 +1,148 @@
+# Spark Rapids Tools End-to-End Behavior Tests
+
+This document outlines the end-to-end tests for Spark Rapids tools, designed to cover scenarios such as missing
+dependencies, handling different types of event logs, and interacting with HDFS.
+
+## Directory Structure
+```commandline
+user_tools/tests/spark_rapids_tools_e2e/
+├── behave.ini                      # Configuration file for the `behave` test runner.
+├── features                        # Contains test scenarios and environment setup.
+│  ├── environment.py               # Setup and teardown procedures for the tests.
+│  ├── steps                        # Step definitions for the tests.
+│  └── *.feature                    # Feature files defining test scenarios.
+└── resources                       # Resources used in the tests.
+    ├── event_logs  
+    └── scripts                     # Scripts used in the tests.  
+```
+
+
+## Setup
+
+From the `<repo_root>/user_tools` directory, run the following command to install the required dependencies:
+
+
+```sh
+pip install behave
+# or
+pip install .[test]
+```
+
+
+## Running Tests
+Tests can be run using 'behave' cmd or using 'tox' cmd.
+
+**Basic Usage:**
+
+```sh
+behave <options>
+# or
+tox -e behave -- <options>
+```
+
+**Run All Tests:**
+
+```sh
+behave
+# or
+tox -e behave
+```
+
+### Common Options
+
+**Run Specific Tests by Tag**
+
+See the [Tags Format](#tags-format) section for more information on tags.
+
+```sh
+behave --tags <tag>
+# or
+tox -e behave -- --tags <tag>
+```
+
+**Run Specific Tests by Name**
+
+```sh
+behave --name <scenario_name>
+# or
+tox -e behave -- --name <scenario_name>
+```
+
+**Skip Tests by Tag**
+
+```sh
+behave --tags ~<tag>
+# or
+tox -e behave -- --tags ~<tag>
+```
+
+**Custom Arguments**
+- Custom arguments can be passed to the behave tests using the `-D` flag.
+- Example: Skip building the Tools jar during setup.
+
+```sh
+behave -D build_jar=false   # Skip building the Tools jar during setup (default: true)
+# or
+tox -e behave -- -D build_jar=false
+```
+
+
```

Comment:
anyway to change debug levels of tools or does it capture the test output in any directories?

Reply:
Added a verbose mode. It can be enabled as follows:
```sh
behave -v
# or
tox -e behave -- -v
```

Updated the docs.